src/views/addPaymentMethod/PaymentMethodForm.tsx

```diff
@@ -1,49 +1,134 @@
+import { API_URL } from '@env'
+import { useQuery } from '@tanstack/react-query'
+import { useCallback, useState } from 'react'
+import { useForm } from 'react-hook-form'
 import { View } from 'react-native'
 import { Header, Screen } from '../../components'
 import { HeaderIcon } from '../../components/Header'
 import { PeachScrollView } from '../../components/PeachScrollView'
 import { Button } from '../../components/buttons/Button'
-import { PaymentMethodForms } from '../../components/inputs/paymentMethods/paymentForms'
-import { useSubmitForm } from '../../components/inputs/paymentMethods/paymentForms/hooks/useSubmitForm'
+import { CurrencySelection } from '../../components/inputs/paymentForms/components'
+import { toggleCurrency } from '../../components/inputs/paymentForms/utils'
 import { useDeletePaymentMethod } from '../../components/payment/hooks/useDeletePaymentMethod'
 import { useRoute, useShowHelp } from '../../hooks'
+import { useGoToOrigin } from '../../hooks/useGoToOrigin'
+import { PAYMENTMETHODINFOS } from '../../paymentMethods'
+import { useOfferPreferences } from '../../store/offerPreferenes'
+import { usePaymentDataStore } from '../../store/usePaymentDataStore'
 import tw from '../../styles/tailwind'
+import fetch from '../../utils/fetch'
 import i18n from '../../utils/i18n'
 import { headerIcons } from '../../utils/layout'
-import { usePaymentMethodFormSetup } from './hooks'
+import { parseResponse } from '../../utils/peachAPI/parseResponse'
+import { getPrivateHeaders } from '../../utils/peachAPI/private/getPrivateHeaders'
+import { PaymentFieldTypes } from '../../utils/validation/rules'
+import { FormInput } from './FormInput'
+import { NewLabelInput } from './NewLabelInput'
+import { TabbedFormNavigation } from './TabbedFormNavigation'
 
-export type FormProps = {
-  data: Partial<PaymentData> & {
-    type: PaymentMethod
-    currencies: Currency[]
-  }
-  onSubmit: (data: PaymentData) => void
-  setStepValid: Dispatch<SetStateAction<boolean>>
-  setFormData: Dispatch<SetStateAction<PaymentData | undefined>>
-}
+// todo: double check cbu, cvu. was called accountNumber before
+// todo: add migration for new field names
+
+export type FormType = Record<PaymentFieldTypes, string> & { paymentMethodName: string }
 
 export const PaymentMethodForm = () => {
-  const { data, onSubmit } = usePaymentMethodFormSetup()
-  const { type: paymentMethod } = data
-  const { stepValid, setStepValid, setFormData, submitForm } = useSubmitForm<PaymentData>(onSubmit)
+  const { paymentData, origin } = useRoute<'paymentMethodForm'>().params
+  const goBackTo = useGoToOrigin()
+  const selectPaymentMethod = useOfferPreferences((state) => state.selectPaymentMethod)
+  const addPaymentData = usePaymentDataStore((state) => state.addPaymentData)
 
-  const Form = PaymentMethodForms[paymentMethod]
+  const { currencies, type: paymentMethod, id, country, label } = paymentData
+
+  const fields = useFormFields(paymentMethod)
```

Comment:
Go through all setup hooks again and compare the field names to the ones used now. add migration or rename fields